plugins/org.jkiss.dbeaver.ui.editors.image/src/org/jkiss/dbeaver/ui/controls/imageview/BrowserImageViewer.java

```diff
@@ -0,0 +1,141 @@
+ */
+public class BrowserImageViewer extends AbstractImageViewer {
+    private static final Log log = Log.getLog(BrowserImageViewer.class);
+
+    private final Browser browser;
+
+    private volatile boolean browserCreating = false;
+    private Path tempFile;
+    private ImageData imageData;
+
+    public BrowserImageViewer(Composite parent, int style) {
+        super(parent, style);
+
+        GridLayout gl = new GridLayout(1, false);
+        gl.horizontalSpacing = 0;
+        gl.verticalSpacing = 0;
+        gl.marginHeight = 0;
+        gl.marginWidth = 0;
+        setLayout(gl);
+        int i = 0;
+
+        browserCreating = true;
+        try {
+            browser = new Browser(this, SWT.NONE);
+        } finally {
+            browserCreating = false;
+        }
+        browser.setLayoutData(new GridData(GridData.FILL_BOTH));
+        browser.setJavascriptEnabled(false); // We don't need java script to open images
+
+
+        // Add DND support
+
+
+    }
+
+    @Override
+    public boolean loadImage(InputStream inputStream) {
+        try {
+            ImageInputStream stream = ImageIO.createImageInputStream(inputStream);
+            Iterator<ImageReader> readers = ImageIO.getImageReaders(stream);
+            if (!readers.hasNext()) {
+                return false;
+            }
+            ImageReader reader = readers.next();
+            reader.setInput(stream);
+
+            tempFile = Files.createTempFile("image", "." + reader.getFormatName().toLowerCase(Locale.ROOT));
+            ImageIO.write(ImageIO.read(stream), reader.getFormatName().toLowerCase(Locale.ROOT), tempFile.toFile());
+            URL url = tempFile.toUri().toURL();
+            browser.setUrl(url.toString());
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+        return true;
+    }
+
+    @Override
+    public boolean clearImage() {
+        browser.setUrl(null);
+        return true;
+    }
+
+    @Override
+    public void dispose() {
+        if (tempFile != null) {
+            try {
+                Files.delete(tempFile);
+            } catch (IOException e) {
+                log.warn(e);
+            }
+            tempFile = null;
+        }
+        // Edge uses callbacks which have a lowest priority in UI
+        // So if dispose is sent during that operation this will lead to initialization
+        // on already disposed composite, we don't want this at all
+        // We can prevent this error by delaying dispose in independent thread operation to allow Edge to finish its
+        // initialization to be disposed properly
+        //FIXME That should be removed as soon as Edge will be fixed, this is an awfull hack
+        if (browser != null) {
+            super.dispose();
+        } else {
+            new Job("Disposing browser") {
+                @Override
+                protected IStatus run(IProgressMonitor monitor) {
+                    while (browserCreating) {
```

Comment:
`browserCreating` will always be `false` here because it's assigned inside constructor.

Also inconsistent naming

Reply:
Please read comments before the method